ethcore/sync/src/light_sync/mod.rs

```diff
@@ -275,6 +300,13 @@ impl<L: AsLightClient + Send + Sync> Handler for LightSync<L> {
 	}
 
 	fn on_disconnect(&self, ctx: &EventContext, unfulfilled: &[ReqId]) {
+		match *self.state.lock() {
```

Comment:
could be an `if let`, right?

Reply:
No, there is a slight difference due to borrowing rules: [play link](http://play.rust-lang.org/?gist=b1d4378f8e0f0ac1a02028365e24b524&version=stable&mode=debug&edition=2015).